crates/bevy_sprite/src/picking_backend.rs

```diff
@@ -135,12 +168,44 @@ pub fn sprite_picking(
 
                 let is_cursor_in_sprite = rect.contains(cursor_pos_sprite);
 
-                blocked = is_cursor_in_sprite
+                let cursor_in_valid_pixels_of_sprite = match settings.picking_mode {
+                    SpritePickingMode::AlphaThreshold(cutoff) if is_cursor_in_sprite => {
+                        let texture: &Image = images.get(&sprite.image)?;
+                        // If using a texture atlas, grab the offset of the current sprite index. (0,0) otherwise
+                        let texture_rect = sprite
+                            .texture_atlas
+                            .as_ref()
+                            .and_then(|atlas| {
+                                texture_atlas_layout
+                                    .get(&atlas.layout)
+                                    .map(|f| f.textures[atlas.index])
+                            })
+                            .or(Some(URect::new(0, 0, texture.width(), texture.height())))?;
+                        // get mouse position on texture
+                        let texture_position = (texture_rect.center().as_vec2()
+                            + cursor_pos_sprite.trunc())
```

Comment:
We need to multiply the cursor_pos_sprite.y by -1. World coordinates go right and up. Image coordinates go right and down. This is why your gif looks a little wonky at the edges of the character sprite (if you zoom in really far).

Reply:
From what I can see, we already convert from world coords to sprite coords above on line 145?

Reply:
Doing some testing, and if you do invert the y value here it leads to frequent out of range errors from bevy image. which leads me further into thinking this isn't the case. happy to be shown wrong though.

Reply:
I was printing the color and it was sampling the wrong pixel for me until I multiplied by -1. One way to test this is to just use a sprite that is opaque on the top half and transparent on the bottom half. Without the -1 you'll be able to pick the transparent half rather than the opaque half!

Reply:
I'll take another look tomorrow, because if so then I also need to figure out the out or bounds issue

Reply:
Ok so, it seems that this -1 issue is actually a wider issue with sprite picking. When the check is done to see if the cursor is within the rect, the position needs to have been multiplied by -1 on the y axis before that point.

`let is_cursor_in_sprite = rect.contains(cursor_pos_sprite);`

The out of range is caused by the uncorrected y position indicating it's within the rect leading to us here trying to get the pixel data for a pixel that doesn't exist within my change. If we multiply by -1 on the y axis before the rect check, not only does it look like it's working more accurately but also the out of range errors seem to stop.

Reply:
Those bevy logos seem off though. Going to keep digging.